apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/service/DexOrderProcessor.java

```diff
@@ -504,6 +518,72 @@ void processCancelOrders(Long accountId) {
         }
     }
 
+    void processExpiredOrders(Long accountId) {
```

Comment:
Looks like this code is almost similar to `processCancelOrders`, i think, that could be better to create general algorithm and customize it for different needs.

Reply:
Add rollback for your cache in `onRollback`

Reply:
Good idea to refactoring it, but I think it will be better to do after our release.